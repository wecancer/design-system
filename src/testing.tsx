import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
