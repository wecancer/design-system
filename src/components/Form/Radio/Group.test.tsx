import RadioGroup from './Group'
import {render, fireEvent} from '../../../testing'

describe('<RadioGroup />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <RadioGroup
          line="column"
          items={[
            {
              id: '1',
              isChecked: false,
              label: 'Item unchecked',
              isDisabled: false,
            },
            {
              id: '2',
              isChecked: true,
              label: 'Item checked',
              isDisabled: false,
            },
            {
              id: '3',
              isChecked: true,
              label: 'Item checked disabled',
              isDisabled: true,
            },
            {
              id: '4',
              isChecked: false,
              label: 'Item unchecked disabled',
              isDisabled: true,
            },
          ]}
        />
        <RadioGroup
          line="column"
          items={[
            {
              id: '1',
              isChecked: false,
              label: 'Item unchecked',
              isDisabled: false,
            },
            {
              id: '2',
              isChecked: true,
              label: 'Item checked',
              isDisabled: false,
            },
          ]}
        />
        <RadioGroup
          line="row"
          items={[
            {
              id: '1',
              isChecked: false,
              label: 'Item unchecked',
              isDisabled: false,
            },
            {
              id: '2',
              isChecked: true,
              label: 'Item checked',
              isDisabled: false,
            },
          ]}
        />
      </>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render all radio group disabled', () => {
    const items = [
      {id: '1', isChecked: false, label: 'Item unchecked', isDisabled: false},
      {id: '2', isChecked: true, label: 'Item checked', isDisabled: false},
      {
        id: '3',
        isChecked: true,
        label: 'Item checked disabled',
        isDisabled: true,
      },
      {
        id: '4',
        isChecked: false,
        label: 'Item unchecked disabled',
        isDisabled: false,
      },
    ]
    const {getAllByRole} = render(
      <RadioGroup line="row" items={items} isDisabled />,
    )
    const radios = getAllByRole('radio')
    radios.forEach((radio) =>
      expect(radio.getAttribute('aria-disabled')).toBe('true'),
    )
  })

  it('should returns valid parameters on change event', () => {
    const items = [
      {id: '1', isChecked: false, label: 'Item 1'},
      {id: '2', isChecked: false, label: 'Item 2'},
      {id: '3', isChecked: false, label: 'Item 3'},
      {id: '4', isChecked: false, label: 'Item 4'},
    ]

    const radioClickIndex = 1

    const {getAllByRole} = render(
      <RadioGroup
        line="row"
        items={items}
        isDisabled
        onChange={(params) => {
          expect(params.index).toBe(radioClickIndex)
          expect(params.item).toMatchObject(items[radioClickIndex])

          expect(params.itemsUpdated[0]).toMatchObject(items[0])
          expect(params.itemsUpdated[1]).toMatchObject({
            ...items[1],
            isChecked: true,
          })
          expect(params.itemsUpdated[2]).toMatchObject(items[2])
          expect(params.itemsUpdated[3]).toMatchObject(items[3])
        }}
      />,
    )

    const component = getAllByRole('radio')[radioClickIndex] as HTMLElement
    fireEvent.click(component)
  })
})
