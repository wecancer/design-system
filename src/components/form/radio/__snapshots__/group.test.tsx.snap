// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should render snapshot correctly 1`] = `
<div>
  <div
    class="sc-hKwDye bVcDfn"
    role="radiogroup"
  >
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="false"
        aria-disabled="false"
        class="sc-gsDKAQ kAIVTa"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item unchecked
      </label>
    </div>
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="true"
        aria-disabled="false"
        class="sc-gsDKAQ eTNswP"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item checked
      </label>
    </div>
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="true"
        aria-disabled="true"
        class="sc-gsDKAQ eTNswP"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item checked disabled
      </label>
    </div>
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="false"
        aria-disabled="true"
        class="sc-gsDKAQ kAIVTa"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item unchecked disabled
      </label>
    </div>
  </div>
  <div
    class="sc-hKwDye bVcDfn"
    role="radiogroup"
  >
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="false"
        aria-disabled="false"
        class="sc-gsDKAQ kAIVTa"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item unchecked
      </label>
    </div>
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="true"
        aria-disabled="false"
        class="sc-gsDKAQ eTNswP"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item checked
      </label>
    </div>
  </div>
  <div
    class="sc-hKwDye jXkjUP"
    role="radiogroup"
  >
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="false"
        aria-disabled="false"
        class="sc-gsDKAQ kAIVTa"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item unchecked
      </label>
    </div>
    <div
      class="sc-bdvvtL gowonf"
    >
      <div
        aria-checked="true"
        aria-disabled="false"
        class="sc-gsDKAQ eTNswP"
        role="radio"
        tabindex="0"
      />
      <label
        class="sc-dkPtRN ebgbMx"
      >
        Item checked
      </label>
    </div>
  </div>
</div>
`;
